### PR TITLE
fixed implicit conversion from ‘float’ to ‘double’ warning

### PR DESCRIPTION
**Pull request by StefanJVA**
fixed implicit conversion from ‘float’ to ‘double’ warning.

An other option could be to change all float variables to double.

### CODE REVIEW

File: ros_ws/src/autonomous/src/wall_following.cpp [hunk 1]
```diff
@@ -76,8 +76,8 @@ void WallFollowing::followSingleWall(const sensor_msgs::LaserScan::ConstPtr& lid
                                     createColor(1, 0, 0, 1), 0.03);
     float wallAngle = wall.getAngle();
     this->m_debug_geometry.drawLine(2, createPoint(PREDICTION_DISTANCE, -error * leftRightSign, 0),
-                                    createPoint(PREDICTION_DISTANCE + cos(wallAngle) * 2,
-                                                (-error + sin(wallAngle) * 2) * leftRightSign, 0),
+                                    createPoint(PREDICTION_DISTANCE + (float)cos(wallAngle) * 2,
+                                                (-error + (float)sin(wallAngle) * 2) * leftRightSign, 0),
```
Comment (marian42) [inline ros_ws/src/autonomous/src/wall_following.cpp]: I don't understand why this is needed. Since `wallAngle` is already a float, shouldn't `sin(wallAngle)` also be a float? Is this related to the literal `2`? Does using `2f` behave differently`?

File: ros_ws/src/autonomous/src/wall_following.cpp [hunk 2]
```diff
@@ -92,7 +92,7 @@ void WallFollowing::followWalls(const sensor_msgs::LaserScan::ConstPtr& lidar)
     float correction = this->m_pid_controller.updateAndGetCorrection(error, TIME_BETWEEN_SCANS);
 
     float steeringAngle = atan(correction) * 2 / M_PI;
-    float velocity = WALL_FOLLOWING_MAX_SPEED * (1 - std::max(0.0, std::abs(steeringAngle) - 0.15));
+    float velocity = WALL_FOLLOWING_MAX_SPEED * (1 - (float)std::max(0.0, (double)std::abs(steeringAngle) - 0.15));
```
Comment (marian42) [inline ros_ws/src/autonomous/src/wall_following.cpp]: Again, I don't understand this. Is there a reasoning why a conversion to double is needed? It looks to me like all numbers in this line are floats. Can this be done without any conversions?

File: ros_ws/src/autonomous/src/wall_following.cpp (hunk 1 above)
Comment (StefanJVA) [reply ros_ws/src/autonomous/src/wall_following.cpp]: Well when i look at this: http://www.cplusplus.com/reference/cmath/sin/
it seems like you are right for c++11. However i always get a warning that says "implicit conversion from ‘float’ to ‘double’". Maybe this is an compiler issue?

Comment (StefanJVA) [reply ros_ws/src/autonomous/src/wall_following.cpp]: @Dadie we need your c++ wizard skillz :D

Comment (Dadie) [reply ros_ws/src/autonomous/src/wall_following.cpp]: In this case the calls to `sin` and `cos` are correctly typed `double`, as those functions are defined only for `double` in the C-Header `math.h` which is included by default by the C++ Wrapper `cmath`.

To call the C++ versions either add the namespace `std::` or ~~enrich~~ pollute the current namespace with `using std;` (**please** don't).

e.g.
```C++
constexpr float CONST_SOMETHING = 1337;
auto should_be_float = std::sin(2.0f) * 2 + CONST_SOMETHING;
```

File: ros_ws/src/autonomous/src/wall.cpp [hunk 3]
```diff
@@ -10,19 +10,19 @@ Wall::Wall(float angle1, float angle2, float range1, float range2)
 
 float Wall::getAngle()
 {
-    return atan((m_range1 * cos(std::abs(m_angle1 - m_angle2)) - m_range2) / m_range1 *
-                sin(std::abs(m_angle1 - m_angle2)));
+    return atan((m_range1 * (float)cos(std::abs(m_angle1 - m_angle2)) - m_range2) / m_range1 *
```
Comment (Dadie) [inline ros_ws/src/autonomous/src/wall.cpp]:
```suggestion
    return std::atan((m_range1 * std::cos(std::abs(m_angle1 - m_angle2)) - m_range2) / m_range1 *
```

File: ros_ws/src/autonomous/src/wall.cpp [hunk 4]
```diff
@@ -10,19 +10,19 @@ Wall::Wall(float angle1, float angle2, float range1, float range2)
...
+                (float)sin(std::abs(m_angle1 - m_angle2)));
```
Comment (Dadie) [inline ros_ws/src/autonomous/src/wall.cpp]:
```suggestion
                std::sin(std::abs(m_angle1 - m_angle2)));
```

File: ros_ws/src/autonomous/src/wall.cpp (hunk 3 above)
Comment (Dadie) [reply ros_ws/src/autonomous/src/wall.cpp]: Because `atan(double)` from `math.h` is used, by replacing it with the C++ one from `cmath` (which is defined for `float`) no conversion from `double` to `float` should happen.

File: ros_ws/src/autonomous/src/wall.cpp (hunk 4 above)
Comment (Dadie) [reply ros_ws/src/autonomous/src/wall.cpp]: Because `sin(double)` from `math.h` is used, by replacing it with the C++ one from `cmath` (which is defined for `float`) no conversion from `double` to `float` should happen.

File: ros_ws/src/autonomous/src/wall.cpp [hunk 5]
```diff
@@ -10,19 +10,19 @@ Wall::Wall(float angle1, float angle2, float range1, float range2)
...
 }
 
 float Wall::predictDistance(float distancce_to_current_position)
 {
     float angle = this->getAngle();
-    float currentWallDistance = m_range2 * cos(angle);
-    return currentWallDistance + distancce_to_current_position * sin(angle);
+    float currentWallDistance = m_range2 * (float)cos(angle);
```
Comment (Dadie) [inline ros_ws/src/autonomous/src/wall.cpp]:
```suggestion
    float currentWallDistance = m_range2 * std::cos(angle);
```
Because `cos(double)` from `math.h` is used, by replacing it with the C++ one from `cmath` (which is defined for `float`) no conversion from `double` to `float` should happen.

File: ros_ws/src/autonomous/src/wall.cpp [hunk 6]
```diff
@@ -10,19 +10,19 @@ Wall::Wall(float angle1, float angle2, float range1, float range2)
...
+    return currentWallDistance + distancce_to_current_position * (float)sin(angle);
```
Comment (Dadie) [inline ros_ws/src/autonomous/src/wall.cpp]:
```suggestion
    return currentWallDistance + distancce_to_current_position * std::sin(angle);
```
Because `sin(double)` from `math.h` is used, by replacing it with the C++ one from `cmath` (which is defined for `float`) no conversion from `double` to `float` should happen.

File: ros_ws/src/autonomous/src/wall.cpp [hunk 7]
```diff
@@ -10,19 +10,19 @@ Wall::Wall(float angle1, float angle2, float range1, float range2)
...
 }
 
 void Wall::draw(RvizGeometryPublisher& geometry, int id, std_msgs::ColorRGBA color)
 {
-    geometry.drawLine(id, createPoint(m_range1 * cos(m_angle1), m_range1 * sin(m_angle1), 0.0),
-                      createPoint(m_range2 * cos(m_angle2), m_range2 * sin(m_angle2), 0.0), color, 0.03);
+    geometry.drawLine(id, createPoint(m_range1 * (float)cos(m_angle1), m_range1 * (float)sin(m_angle1), 0.0),
```
Comment (Dadie) [inline ros_ws/src/autonomous/src/wall.cpp]:
```suggestion
    geometry.drawLine(id, createPoint(m_range1 * std::cos(m_angle1), m_range1 * std::sin(m_angle1), 0.0f),
```
Because `cos(double)` from `math.h` is used, by replacing it with the C++ one from `cmath` (which is defined for `float`) no conversion from `double` to `float` should happen. Also added the `float` literal to the floating number.

File: ros_ws/src/autonomous/src/wall.cpp [hunk 8]
```diff
@@ -10,19 +10,19 @@ Wall::Wall(float angle1, float angle2, float range1, float range2)
...
+                      createPoint(m_range2 * (float)cos(m_angle2), m_range2 * (float)sin(m_angle2), 0.0), color, 0.03);
```
Comment (Dadie) [inline ros_ws/src/autonomous/src/wall.cpp]:
```suggestion
                      createPoint(m_range2 * std::cos(m_angle2), m_range2 * std::sin(m_angle2), 0.0f), color, 0.03);
```
Because `cos(double)` and `sin(double)` from `math.h` is used, by replacing it with the C++ one from `cmath` (which is defined for `float`) no conversion from `double` to `float` should happen. Also added the `float` literal to the floating number.

File: ros_ws/src/autonomous/src/wall_following.cpp [hunk 9]
```diff
@@ -76,8 +76,8 @@ void WallFollowing::followSingleWall(const sensor_msgs::LaserScan::ConstPtr& lid
                                     createColor(1, 0, 0, 1), 0.03);
     float wallAngle = wall.getAngle();
     this->m_debug_geometry.drawLine(2, createPoint(PREDICTION_DISTANCE, -error * leftRightSign, 0),
-                                    createPoint(PREDICTION_DISTANCE + cos(wallAngle) * 2,
-                                                (-error + sin(wallAngle) * 2) * leftRightSign, 0),
+                                    createPoint(PREDICTION_DISTANCE + (float)cos(wallAngle) * 2,
```
Comment (Dadie) [inline ros_ws/src/autonomous/src/wall_following.cpp]:
```suggestion
                                    createPoint(PREDICTION_DISTANCE + std::cos(wallAngle) * 2,
```
Because `cos(double)` from `math.h` is used, by replacing it with the C++ one from `cmath` (which is defined for `float`) no conversion from `double` to `float` should happen.

File: ros_ws/src/autonomous/src/wall_following.cpp (hunk 2 above)
Comment (Dadie) [inline ros_ws/src/autonomous/src/wall_following.cpp]:
```suggestion
    float velocity = WALL_FOLLOWING_MAX_SPEED * (1 - std::max(0.0f, std::abs(steeringAngle) - 0.15f));
```
Added the `float` literal to the floating number, which should fix the warning.

File: ros_ws/src/autonomous/src/wall_following.cpp (hunk 1 above)
Comment (Dadie) [inline ros_ws/src/autonomous/src/wall_following.cpp]:
```suggestion
                                                (-error + std::sin(wallAngle) * 2) * leftRightSign, 0),
```
Because `sin(double)` from `math.h` is used, by replacing it with the C++ one from `cmath` (which is defined for `float`) no conversion from `double` to `float` should happen.
